internal/command/meta_backend.go

```diff
@@ -670,30 +670,15 @@ func (m *Meta) backendFromConfig(opts *BackendOpts) (backend.Backend, tfdiags.Di
 		}
 		log.Printf("[TRACE] Meta.Backend: backend configuration has changed (from type %q to type %q)", s.Backend.Type, c.Type)
 
-		initReason := ""
-		switch {
-		case c.Type == "cloud":
-			initReason = fmt.Sprintf("Backend configuration changed from %q to Terraform Cloud", s.Backend.Type)
-		case s.Backend.Type != c.Type:
-			initReason = fmt.Sprintf("Backend configuration changed from %q to %q", s.Backend.Type, c.Type)
-		default:
-			initReason = fmt.Sprintf("Backend configuration changed for %q", c.Type)
-		}
-
 		if !opts.Init {
-			if c.Type == "cloud" {
-				diags = diags.Append(tfdiags.Sourceless(
-					tfdiags.Error,
-					"Terraform Cloud initialization required, please run \"terraform init\"",
-					fmt.Sprintf(strings.TrimSpace(errBackendInitCloud), initReason),
-				))
-			} else {
-				diags = diags.Append(tfdiags.Sourceless(
-					tfdiags.Error,
-					"Backend initialization required, please run \"terraform init\"",
-					fmt.Sprintf(strings.TrimSpace(errBackendInit), initReason),
-				))
-			}
+			//user ran another cmd that is not init but they are required to initialize because of a potential relevant change to their backend configuration
+			initReason := m.determineInitReason(s.Backend.Type, c.Type)
+
+			diags = diags.Append(tfdiags.Sourceless(
+				tfdiags.Error,
+				"Backend initialization required, please run \"terraform init\"",
+				fmt.Sprintf(strings.TrimSpace(errBackendInit), initReason),
```

Comment:
To be consistent about avoiding the "backend" terminology when we're using the Cloud integration, I think we should create a different diagnostic here: one with the `"Terraform Cloud initialization required, …"` summary, and the `errBackendInitCloud` template. I think that would mean undoing the removal of the `if c.Type == "cloud"` conditional above.

Was there a reason to switch to always using this `"Backend initialization required, …"` diagnostic that I missed?

Reply:
Whoa, I totally didn't realize that I had change it from `"Terraform Cloud initialization required, …"`  to `"Backend initialization required, …"` when I removed the `if c.Type == "cloud"` conditional above! My bad. Thanks for making me notice.